src/main.ts

```diff
@@ -20,32 +20,35 @@ async function run() {
       {
         const environment = core.getInput('env', { required: true });
         const noOverride = core.getInput('no_override') === 'true';
-        console.log(`initializing deployment for ${environment}`);
+        let deploymentID = core.getInput('deployment_id', { required: false });
+        console.log(`initializing deployment ${deploymentID} for ${environment}`);
 
         // mark existing deployments of this environment as inactive
         if (!noOverride) {
           await deactivateEnvironment(client, repo, environment);
         }
 
         const transient = core.getInput('transient', { required: false }) === 'true';
-        const deployment = await client.repos.createDeployment({
-          owner: repo.owner,
-          repo: repo.repo,
-          ref: ref,
-          required_contexts: [],
-          environment,
-          auto_merge: false,
-          transient_environment: transient,
-        });
+        if ( ! deploymentID ) {
```

Comment:
nit, but for consistency:

```suggestion
        if (!deploymentID) {
```